include/oneapi/dpl/pstl/hetero/dpcpp/sycl_iterator.h

```diff
@@ -110,89 +104,81 @@ struct sycl_iterator
         return *this - it < 0;
     }
 
-    sycl::buffer<T, dim, Allocator>
+    Buffer
     get_buffer() const
     {
         return buffer;
     }
 };
-
-// mode converter when property::noinit present
-template <access_mode __mode>
-struct _ModeConverter
-{
-    static constexpr access_mode __value = __mode;
-};
-
-template <>
-struct _ModeConverter<access_mode::read_write>
-{
-    static constexpr access_mode __value = access_mode::discard_read_write;
-};
-
-template <>
-struct _ModeConverter<access_mode::write>
-{
-    static constexpr access_mode __value = access_mode::discard_write;
-};
-
 } // namespace __internal
 
+// begin
 template <typename T, typename Allocator>
-__internal::sycl_iterator<access_mode::read_write, T, Allocator> begin(sycl::buffer<T, /*dim=*/1, Allocator> buf)
+__internal::sycl_iterator<T, Allocator>
+begin(sycl::buffer<T, /*dim=*/1, Allocator> buf)
 {
-    return __internal::sycl_iterator<access_mode::read_write, T, Allocator>{buf, 0};
+    return {buf, 0};
 }
 
+// end
 template <typename T, typename Allocator>
-__internal::sycl_iterator<access_mode::read_write, T, Allocator> end(sycl::buffer<T, /*dim=*/1, Allocator> buf)
+__internal::sycl_iterator<T, Allocator>
+end(sycl::buffer<T, /*dim=*/1, Allocator> buf)
 {
-    return __internal::sycl_iterator<access_mode::read_write, T, Allocator>{buf, __dpl_sycl::__get_buffer_size(buf)};
+    return {buf, __dpl_sycl::__get_buffer_size(buf)};
 }
 
-// begin
-template <typename T, typename Allocator, access_mode Mode>
-__internal::sycl_iterator<Mode, T, Allocator> begin(sycl::buffer<T, /*dim=*/1, Allocator> buf, sycl::mode_tag_t<Mode>)
+// Old variants of begin/end for compatibility with old code
+inline namespace depreated
 {
-    return __internal::sycl_iterator<Mode, T, Allocator>{buf, 0};
-}
+    using access_mode = sycl::access::mode;
 
-template <typename T, typename Allocator, access_mode Mode>
-__internal::sycl_iterator<__internal::_ModeConverter<Mode>::__value, T, Allocator>
+    // begin
+    template <typename T, typename Allocator, access_mode Mode>
+    __internal::sycl_iterator<T, Allocator>
+    begin(sycl::buffer<T, /*dim=*/1, Allocator> buf, sycl::mode_tag_t<Mode>)
+    {
+        return oneapi::dpl::begin(buf);
```

Comment:
I think we have to print a kind of "deprecation" warning on each call of that deprecated  API.

Reply:
Fixed